### PR TITLE
feat: Implement `Bessel_j1`

### PR DESCRIPTION
**Pull request by HarshitaKalani**

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
I tried to implement this. Currently the output is not aligning with GFortran. 
Working on that.

**Review by certik — APPROVED** [review]
This is good. I thought the libc library has implementation of these, but it's good to have explicit implementations anyway, we'll have similar implementations for all math functions eventually, and we'll figure out how to organize it. For now this is good.

**Comment by kargl** [comment]
What is the software license for this piece of code?  It seems that lfortran is now in violation of Copyright.

**Comment by certik** [comment]
@kargl it's Apache 2.0 and Boost. We just need to copy the licenses, which I just did in https://github.com/lfortran/lfortran/pull/3696.

**Comment by kargl** [comment]
> @kargl it's Apache 2.0 and Boost. We just need to copy the licenses, which I did in #3696.

Yes, I read the code and tracked down the licenses.  The original commit appears to violate copyright.  It's unclear to me whether displacing the licenses from the code is acceptable under copyright.

The question now is what other code in lfortran may have been imported from other projects without proper license(s)?

**Comment by certik** [comment]
@kargl we diligently track all 3rd party licenses and which parts they apply to at https://github.com/lfortran/lfortran/blob/main/LICENSE. Thank you for reminding me about the above. We should have included it LICENSE right away, however, we did mark the parts of code that we copied. We develop LFortran completely in the open via pull requests, so it's easy to track and audit any line of code that goes in (which PR it came from, who wrote it, the review, the discussion around it, etc.).

If you discover any other contribution that is not properly listed there, definitely let me know.

> It's unclear to me whether displacing the licenses from the code is acceptable under copyright.

This is a good question. I think it's allowed, for example the Boost license says _"The copyright notices in the Software and this entire statement, including the above license grant, this restriction and the following disclaimer, must be included in all copies of the Software, in whole or in part, and all derivative works of the Software"_. Our LICENSE file does exactly that. In fact, some licenses do not require to do this for binary distributions (for example this Boost license says _"... unless such copies or derivative works are solely in the form of machine-executable object code generated by a source language processor."_), but we copy LICENSE in our binary distribution also (the Conda package distributes the LICENSE file).

> The question now is what other code in lfortran may have been imported from other projects without proper license(s)?

None that I know of, all our contributors write the code themselves, and if we copy code, we always mark it as such, including in this PR.

If you or anybody else discovers any issue, we'll fix it.

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -1208,6 +1207,222 @@ LFORTRAN_API float _lfortran_sbesselj0( float x ) {
...
+    if ( x < 0.0 ) {
+        value = -1.0 * value;
+    }
+    return value;
+}
+
+LFORTRAN_API float _lfortran_sbesselj1( float x ) {
+    return (float)_lfortran_dbesselj1((double)x);
+}
+
```
Comment (Shaikh-Ubaid) [inline src/libasr/runtime/lfortran_intrinsics.c:1425]: For now this seems fine. Our long-term plan is to remove the C runtime library completely. From the long-term vision it seems we will have to refactor the above code.

Comment (Shaikh-Ubaid) [reply src/libasr/runtime/lfortran_intrinsics.c:1425]: Just noticed here that this code was removed https://github.com/lfortran/lfortran/pull/3737. That is great.

**Pull request merged**